jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_annotations.cpp

```diff
@@ -91,5 +91,30 @@ SCENARIO(
         REQUIRE(java_type == void_type());
       }
     }
+    WHEN("Parsing an annotation with an array field.")
+    {
+      const symbol_tablet &new_symbol_table = load_java_class(
+        "ClassWithArrayAnnotation", "./java_bytecode/java_bytecode_parser");
+
+      THEN("The annotation should store an array of type string.")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::ClassWithArrayAnnotation");
+        const std::vector<java_annotationt> &java_annotations =
+          to_annotated_type(class_symbol.type).get_annotations();
```

Comment:
Nit: in `require_types` you can add a method called `require_annotated_type` that would look like:

```C++
annotated_typet require_annotated_type(const typet &type)
{
  REQUIRE(can_cast_type<annotted_typet>(type);
  return to_annotated_typet(type);
}
```

As the above line, if the type isn't `annotated_typet` will abort the unit test making it harder to see what went wrong.